### PR TITLE
fix(user): add conditional rendering for hide completed sets checkbox

### PR DESCRIPTION
**Pull request by wescopeland**
This PR addresses feedback left by @Jamiras in https://github.com/RetroAchievements/RAWeb/pull/1477#pullrequestreview-1392714663 for the "Hide completed sets" checkbox.

With this branch's changes, the new checkbox will no longer appear unless the user has at least 1 completed set and 1 non-completed set.

### CODE REVIEW

File: app_legacy/Helpers/render/user.php [hunk 1]
```diff
@@ -147,23 +147,52 @@ function renderUserCard(string|array $user): string
...
+        $nextMaxPossible = $userCompletedGamesList[$i]['MaxPossible'];
+        $nextNumAwarded = $userCompletedGamesList[$i]['NumAwarded'];
+
+        // Don't divide by 0.
+        if ($nextNumAwarded == 0 || $nextMaxPossible == 0) {
+            continue;
+        }
+
+        $pctAwardedNormal = ($nextNumAwarded / $nextMaxPossible) * 100.0;
+        if ($pctAwardedNormal == 100) {
```
Comment (Jamiras) [inline app_legacy/Helpers/render/user.php:166]: `if ($nextNumAwarded == $nextMaxPossible)`?

Would avoid doing the multiplication and division.

File: app_legacy/Helpers/render/user.php [hunk 2]
```diff
@@ -147,23 +147,52 @@ function renderUserCard(string|array $user): string
     return $tooltip;
 }
 
+function getCompletedAndIncompletedSetsCounts(array $userCompletedGamesList): array
+{
+    $completedSetsCount = 0;
+    $incompletedSetsCount = 0;
+
+    $numItems = count($userCompletedGamesList);
+    for ($i = 0; $i < $numItems; $i++) {
```
Comment (Jamiras) [inline app_legacy/Helpers/render/user.php:156]: Consider using foreach:
```
foreach ($userCompletedGamesList as $game) {
    $nextMaxPossible = $game['MaxPossible']
    ....
```

Avoids doing the count, and two indexed lookups for every iteration.

File: app_legacy/Helpers/render/user.php (hunk 1 above)
Comment (wescopeland) [reply app_legacy/Helpers/render/user.php]: Resolved in https://github.com/RetroAchievements/RAWeb/pull/1492/commits/5353618147739ece716f4f3e6ef5991ddf1d9098.

File: app_legacy/Helpers/render/user.php (hunk 2 above)
Comment (wescopeland) [reply app_legacy/Helpers/render/user.php]: That does simplify it quite a bit. Also resolved in https://github.com/RetroAchievements/RAWeb/pull/1492/commits/5353618147739ece716f4f3e6ef5991ddf1d9098.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**